File: rotate_array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        '''copy arr, rotate that one, copy back'''
        new_arr = nums[:]

        for i in range(k):
            new_arr = [new_arr[-1]] + new_arr
            new_arr.pop()

        for i in range(len(nums)):
            nums[i] = new_arr[i]
        print(nums)

sol = Solution()
sol.rotate([1, 2, 3, 4, 5, 6, 7], 3)
sol.rotate([1], 2)

File: rotateArray.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        vector<int> copy = nums;
        for (int i = 0; i < copy.size(); i++)
            nums[(i+k)%copy.size()] = copy[i];
    }
};

int main() {

    return 0;
}